### PR TITLE
move update into base market

### PR DESCRIPTION
**Pull request by wakamex**

### CODE REVIEW

**Review by wakamex — COMMENTED** [review]
lgtm

File: elfpy/markets/base.py [hunk 1]
```diff
@@ -70,7 +66,7 @@ class MarketTradeResult:
 
 @types.freezable(frozen=False, no_new_attribs=False)
 @dataclass
-class BaseMarketState:
+class BaseMarketState(Generic[State]):
```
Comment (dpaiton) [inline elfpy/markets/base.py:69]: I think you want to remove this

File: elfpy/markets/borrow.py [hunk 2]
```diff
@@ -110,6 +110,13 @@ class MarketState(base_market.BaseMarketState):
     # borrow rate is lending_rate * spread_ratio
     spread_ratio: float = field(default=1.25)
 
+    def __post_init__(self) -> None:
+        r"""Initialize the market state"""
+        # initialize loan to value ratios if a float is passed
+        if isinstance(self.loan_to_value_ratio, float):
+            self.loan_to_value_ratio = {token_type: self.loan_to_value_ratio for token_type in types.TokenType}
+            print(f"LTVs initialized to {self.loan_to_value_ratio=}")
```
Comment (dpaiton) [inline elfpy/markets/borrow.py:118]: no print

**Review by dpaiton — APPROVED** [review]
approved after comments are addressed.

File: elfpy/markets/base.py (hunk 1 above)
Comment (wakamex) [reply elfpy/markets/base.py:69]: works either way, removing it

**Pull request merged**